aten/src/ATen/FunctionalTensorImplBase.cpp

```diff
@@ -0,0 +1,184 @@
+#include <ATen/FunctionalTensorImplBase.h>
+
+#include <ATen/core/List.h>
+#include <c10/util/Exception.h>
+
+namespace at {
+
+FunctionalTensorImplBase::FunctionalTensorImplBase(c10::DispatchKeySet keyset, caffe2::TypeMeta dtype, c10::Device device)
+  : TensorImpl(
+      keyset.add(DispatchKey::Functionalize),
+      dtype,
+      device
+    ) {}
+
+FunctionalTensorImplBase::FunctionalTensorImplBase(caffe2::TypeMeta dtype, c10::Device device)
+  : TensorImpl(
+      c10::DispatchKeySet(DispatchKey::Functionalize),
+      dtype,
+      device
+    ) {}
```

Comment:
Wondering why we need both constructors

Reply:
There are two subclasses (`FunctionalTensorWrapper` for functorch, `XLATensorImpl` for XLA).

XLA (and presumably other backends) will want to specify their own backend key(s) to include in the keyset, while functorch's subclass is just a wrapper, so it only needs the `Functionalize` key - since the DynamicLayer logic will unwrap the subclass and to figure out the next key to dispatch to.

I could just have the wrapper call the first constructor with an empty keyset and kill the second constructor, since it's only really for functorch. LMK what you think.